shortcut.js

```diff
@@ -20,15 +36,59 @@ const query = async (doc, sel) => {
   }
 };
 
+const getPrDetails = pr => {
+  const title = document.querySelector("bdi.js-issue-title.markdown-title").innerText;
+  const commits = [...document.querySelectorAll(".TimelineItem-body a.markdown-title[href*='/commits/']")]
+    .flatMap(({ title, href }) => {
+      const match = /\/NixOS\/nixpkgs\/pull\/(\d+)\/commits\/([0-9a-f]+)$/i.exec(href);
+      return (match === null || match[1] !== pr) ? [] : [{
+        commit_id: match[2],
+        subject: title.split("\n")[0],
+        description: title,
+      }];
+    });
+  const labels = [...document.querySelectorAll("div.js-issue-labels > a")].map(x => x.innerText);
+  const author = document.querySelector(".js-discussion > :first-child a.author").href.split("/").at(-1);
+  const self = document.querySelector("div.AppHeader-user button[data-login]").getAttribute("data-login");
+  const authoredByMe = author === self;
+  const hasLinuxRebuilds = !labels.some(l => /rebuild-linux: 0$/.test(l));
+  const hasDarwinRebuilds = !labels.some(l => /rebuild-darwin: 0$/.test(l));
```

Comment:
How will this handle the scenario where the button is clicked _before_ the PR has been labeled?

Reply:
In that case we cannot easily find out the number of rebuilds, so we simply assume it is greater than zero and enable the builds on the corresponding systems to be on the safe side.